Update dependencies & enable Dependabot version updates

Pl217 opened the pull request.

Pl217 committed:
Add `reinstall` command

Pl217 committed:
Require Yarn version `>=1.22.22`

Pl217 committed:
Update minimum Node version to `v20.15.1`

Pl217 committed:
Update `class-validator` to `v0.14.1`

Pl217 committed:
Update PM2 `v5.4.2`

Pl217 committed:
Update `reflect-metadata` to `v0.2.2`

Pl217 committed:
Update `ts-node` to `v10.9.2`

Pl217 committed:
Update `@unocha/hpc-api-core` to `v8.2.0`

Pl217 committed:
Update to Typescript `v5.5.4`

Pl217 committed:
Update `@unocha/hpc-repo-tools` to `v4.2.0`

Pl217 committed:
Update `eslint` to `v8.57.0`

Pl217 committed:
Update Husky to `v9.1.5`

Code changes made by following migration guides from these two releases:
https://github.com/typicode/husky/releases/tag/v9.0.1
https://github.com/typicode/husky/releases/tag/v9.1.1

Pl217 committed:
Update `lint-staged` to `v15.2.9`

Pl217 committed:
Update Prettier to `v3.3.3`

Pl217 committed:
Update `ts-jest` to `v29.2.5`

Pl217 committed:
Enable Dependabot version updates

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

Pl217 committed:
Use default import style when importing from `node:path`

The current use violates linting rule `unicorn/import-style`
and we're getting error message: "Use default import for module `node:path`"

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review
